MobileCenter/MobileCenter/Internals/AppDelegate/MSAppDelegateForwarder.m

```diff
@@ -179,17 +198,40 @@ + (IMP)swizzleOriginalSelector:(SEL)originalSelector
      * depend on the SDK return value for its own logic so customers already have to call the SDK API
      * in their implementation which makes swizzling useless.
      */
-    if (![self.selectorsNotToOverride containsObject:originalSelectorString]) {
+    if (![self.selectorsNotToOverride containsObject:originalSelectorStr]) {
       originalImp = method_setImplementation(originalMethod, customImp);
+    } else {
+      warningMsg =
+          [NSString stringWithFormat:@"This selector is not supported when already implemented. %@", remediationMsg];
     }
   } else if (![originalClass instancesRespondToSelector:originalSelector]) {
 
-    /*
-     * The original class may not implement the selector (e.g.: optional method from protocol),
-     * add the method to the original class and associate it with the custom implementation.
-     */
-    Method customMethod = class_getInstanceMethod(self, customSelector);
-    methodAdded = class_addMethod(originalClass, originalSelector, customImp, method_getTypeEncoding(customMethod));
+    // Check for deprecation.
+    NSString *deprecatedSelectorStr = self.deprecatedSelectors[originalSelectorStr];
+    if (deprecatedSelectorStr &&
+        [originalClass instancesRespondToSelector:NSSelectorFromString(deprecatedSelectorStr)]) {
+
+      // An implementation for the deprecated selector exists. Don't add the new method, it might eclipse the original
```

Comment:
Multiline comment. Use `/* */`

Reply:
Good catch, it got multilined by the formatter.